digicampipe/image/sky_image.py

```diff
@@ -0,0 +1,555 @@
+                                                         catalog=['I/271/out'])[0]
+                green_stars_ra_dec = np.array(
+                    (result_green['RA_ICRS_'], result_green['DE_ICRS_'])).transpose()  # .reshape((-1,2))
+                green_stars_mag = result_blue['Vmag']
+                green_stars_px = sky_image.wcs.wcs_world2pix(green_stars_ra_dec, 1)
+                for star_px, star_mag in zip(green_stars_px, green_stars_mag):
+                    if -5 < star_mag <= 12:
+                        radius = 18 - star_mag
+                    else:
+                        radius = 5
+                    circle = Circle((star_px[0] + crop_pixel1[0], star_px[1] + crop_pixel1[1]), radius=radius,
+                                    fill=False, color='g')
+                    ax.add_artist(circle)
+                result_gamma = vizier_gamma.query_region(center_coordinate, radius=Angle(5, "deg"),
+                                                         catalog=['J/ApJS/197/34'])[0]
+                gamma_stars_ra_dec = np.array((result_gamma['RAJ2000'], result_gamma['DEJ2000'])).transpose()  # .reshape((-1,2))
+                gamma_stars_name = result_gamma['Name']
+                gamma_stars_px = sky_image.wcs.wcs_world2pix(gamma_stars_ra_dec, 1)
+                for star_px, gamma_star_name in zip(gamma_stars_px, gamma_stars_name):
+                    circle = Circle((star_px[0] + crop_pixel1[0], star_px[1] + crop_pixel1[1]), radius=20,
+                                    fill=False, color='Y')
+                    ax.add_artist(circle)
+                    ax.text(star_px[0] + crop_pixel1[0], star_px[1] + crop_pixel1[1], gamma_star_name, color='Y')
+                crab_nebula_px = sky_image.wcs.wcs_world2pix(83.640187, 22.044295, 1)
+                circle = Circle((crab_nebula_px[0] + crop_pixel1[0], crab_nebula_px[1] + crop_pixel1[1]), radius=20,
+                                fill=False, color='Y')
+                ax.add_artist(circle)
+                ax.text(crab_nebula_px[0] + crop_pixel1[0], crab_nebula_px[1] + crop_pixel1[1], 'Crab Pulsar', color='Y')
+            width = crop_pixel2[0] - crop_pixel1[0]
+            height = crop_pixel2[1] - crop_pixel1[1]
+            rect = Rectangle(crop_pixel1, width=width, height=height, fill=False, color='y', linestyle='dashdot')
+            ax.add_artist(rect)
+        # plt.imshow(image_treated, cmap='gray')
+        plt.plot(0, 0, 'w+')
+        plt.grid(None)
+        plt.axis('off')
+        ax.get_xaxis().set_visible(False)
+        ax.get_yaxis().set_visible(False)
+        if output_dir is None:
+            plt.show()
+        else:
+            output_filename = self.filename.replace('.fits', '-treated.png')
+            plt.savefig(output_filename, bbox_inches='tight', pad_inches=0)
+            plt.close(fig)
+            print(output_filename, 'saved.')
+
+
+class LidCCDObservation:
```

Comment:
why is this a class? It looks like it is actually a function which returns `lidccd_images` the 3 methods this class has, are actually functions, which act on lists lists of ccd_images.

Reply:
Several images are needed to detect stars (looking for movement).
The only specific work done by LidCCDObservation is to calculate the average of all images, that average being then used to find stars.